### ISSUE TITLE
Allow separate windows to have separate zoom settings

### ISSUE THREAD

**Issue by raffi23** [microsoft/vscode#143663]
I use multiple monitors each has different DPIs and it is really annoying that I can't have different levels of zoom on each window I have opened, so at the moment smaller screens have smaller font sizes automatically.

I have seen multiple issues open on this regard, some was disregarded and some was rejected even though it had more than 20 upvotes like [#72011](https://github.com/Microsoft/vscode/issues/72011)

I am pretty sure that a lot of people will be a lot more productive if this feature was implemented.

**Comment by bpasero** [comment]
The zoom level can be set as a workspace setting.

**Comment by raffi23** [comment]
> The zoom level can be set as a workspace setting.

In my opinion it should not be based on workspace settings it should be window based.

**Comment by kubukoz** [comment]
Yes, workspace isn't specific enough, especially if someone uses multiple windows per workspace.

**Comment by luke-uppx** [comment]
I don't understand why it hasn't been implemented, in IntelliJ IDEA it's that simple.
![image](https://user-images.githubusercontent.com/54334100/159242168-e816c4d4-c333-4b8c-a7ec-45cc2f9b7706.png)

**Comment by Zingam** [comment]
Here is a related issue: https://github.com/microsoft/vscode/issues/146254

Also it would be great to have a zoom level indicator/selector like in VS 2022.

**Comment by PAQTnikola** [comment]
Would it be possible to set a zoom level per PPI? So that if I drag a VS Code window from one monitor to another that has a different PPI that the zoom level then automatically updates.

This is a slightly different feature than different zoom levels per VS Code window, so it might be out of scope.

**Comment by eviathan** [comment]
If I am completely honest this is more than a little inconvenient when you have monitors with wildly different ppis. What would be better is if you could take the ppis into account but failing that the option to have independent scaling per window is almost necessary for making this feature even remotely useful for me.
 
The only reason I (and probably many others) wanted multiple windows was to use them full screen across multiple monitors. When I do that on my 43" 4K monitor and my vertical 27" 4K monitor the text becomes either too large or unreadably small.

This feels like a half arsed implementation. Sorry.

**Comment by CreationStory** [comment]
Add

**Comment by shawnfromportland** [comment]
listen, maybe i want to make the font on all my terminal panes in vscode really  then maybe i want to quickly zoom in the current window editor code itself without affecting the title bar and menu texts of vscode. 
By not allowing me full independent control of all font sizes in each window from title bar font sizes to menu font sizes to editor font sizes to terminal font sizes, you are making the assertion that i as the developer should not be in control of how i use my own screen real estate

**Comment by CreationStory** [comment]
if visual studio has it and notepad++ has it, vscode will never get it

**Comment by electrovir** [comment]
This can be accomplished by modifying your keyboard shortcuts from `View: Zoom In` / `View: Zoom Out` to `Editor: Font Zoom In` / `Editor: Font Zoom Out`.

**Comment by shawnfromportland** [comment]
> This can be accomplished by modifying your keyboard shortcuts from `View: Zoom In` / `View: Zoom Out` to `Editor: Font Zoom In` / `Editor: Font Zoom Out`.

thanks, that's not the same. for example what if i want a huge font for my project file list, small editor font, and medium terminal font. i already control editor font size with ctrl+mouse wheel but i can't independently control other panes

**Comment by lgarron** [comment]
I know this is currently marked as not planned, but I'd really love to see this reopened, perhaps with an invitation for a third-party PR.

Like others here, I regularly use multiple monitors. In particular, I use the built-in laptop monitor on the go, and a 5K monitor at my desk. I use a zoom level of 3 on my monitor, but I have to set it below 2 on my laptop to get anywhere near a comparable balance of how much text is visible.

One thing I haven't seen mentioned here: the existing `window.zoomLevel` setting really messes with anyone who uses dotfiles to manage their settings.[^1]

In particular:

- For the last few years, almost every time I go to manage my dotfiles, I have to explicitly avoid committing `window.zoomLevel` so that I don't commit it together with unrelated changes. It also means that my dotfiles repo is almost always shown as "dirty" by `git`.
- If I *don't* avoid committing `window.zoomLevel`, then I will constantly have to deal with merge conflicts. Since I have two computers on which I manage my dotfiles, I would run into these merge conflicts on more than a daily basis.

The following alternatives would be much, much less cumbersome for me:

- Have an option to set zoom level based on screen resolution. For example:

```
  "window.zoomLevel": {
    "default": 2,
    "byResolution": {
      "2560x1440@2x": 3,
      "1728x1117@2x": 1.75,
    }
  }
```

The `⌘+` and `⌘-` shortcuts would then serialize to whichever setting is currently being used by the screen of the window where the shortcut is pressed.

- Have an option to avoid serializing `window.zoomLevel` in `settings.json`. At this point I'd be willing to manually the zoom every time I start VS Code just to avoid the dotfiles issue, although it would be nice if it was recorded somewhere. This could either be combined with the previous bullet:

```
{
  "window.zoomLevel": {
    "default": 2,
    "serializeSettingsOnChange": "never",
    "byResolution": {
      // …
    }
  }
}
```

Or a simpler version:


```jsonc
{
  "window.zoomLevel": 2,
  "window.zoomLevel.remember": false
}
```

- Provide a way for `window.zoomLevel` to be recorded in a separate file.
  - One potential workaround would be to try to use profile inheritance to maintain one profile of dynamic settings, while the zoom level is saved in a parent profile whose changes are ignored by `git`. But I don't think it's possible to do this in a way that `⌘+` and `⌘-` will write to the ignored file instead of the child profile.

[^1]: I know about settings sync. I use it. However, VS Code clearly believes the power of declarative serialized reactive configuration, so I think this is a reasonable way to use it. [This result](https://github.com/search?q=dotfiles+%22window.zoomLevel%22&type=code&p=4) shows 800 results for people using dotfiles to manage their VS Code settings, and that's just results for `window.zoomLevel` of those published to GitHUb.

**Comment by michalriha1** [comment]
I addressed this by having shortcuts to set a specific zoom for each monitor I regularly use.
```javascript
    {
        "key": "shift+cmd+-",
        "command": "workbench.action.setZoomLevel",
        "args": 0
    },
    {
        "key": "shift+cmd+=",
        "command": "workbench.action.setZoomLevel",
        "args": "0.2"
    }
```

**Comment by Quadrophonia** [comment]
Adding my voice to the request for per-window zoom control. VSCode is great, but can become cumbersome when working across multiple monitors (or even multiple windows in a single monitor). I don't really need to see Terminal output at the same magnification as my code, or my interactive window outputs. It just makes sense to have them at different magnifications. I'm not convinced it would be terribly difficult to do.
